README enhancements

cobbweb opened the pull request:
[Live preview](https://github.com/cobbweb/Vim/tree/feature/improve-readme-structure#------------vscodevim)

A few README enhancements. Mostly re-arranging things and, fixing up formatting and consistency problems, including:

* Add logo to header and center align it
* Add Contents section
* Tighten up Features list
* Add OS-specific setup notes
* Promote plugins to their own section
* Fix up various Markdown and indentation issues
* Add syntax highlighting to JSON examples
* Improve heading consistency (use "sentence case")
* Make settings-based headings look like their JSON key equivalents
* Prefer the language "settings" over "option" or "configuration" where possible.

Sorry for all the commits, happy to squash them once this is good to merge. I'd like to have another pass at proofreading over the weekend too. Let me know if there's anything you want removed/revert/fixed :+1:

Thanks for an awesome plugin too

xconverge commented:
Can you add the recent lightline plugin information? Sorry I think I added that since you branched :)

cobbweb commented:
@xconverge Are you talking about the status bar colors? That's in there, but under Settings. I didn't want to put it under 'Emulated plugins', as the other under emulated plugins have really good feature parity. Happy to move it if you want, but I thought it could be a bit misleading.

jpoon commented on README.md:
```diff
@@ -1,19 +1,35 @@
-# Vim [![Version](http://vsmarketplacebadge.apphb.com/version/vscodevim.vim.svg)](http://aka.ms/vscodevim) [![Build Status](https://travis-ci.org/VSCodeVim/Vim.svg?branch=master)](https://travis-ci.org/VSCodeVim/Vim) [![Slack Status](https://vscodevim-slackin.azurewebsites.net/badge.svg)](https://vscodevim-slackin.azurewebsites.net)
+<h1 align="center">
```
I'm not a big fan of embedding HTML into the markdown

cobbweb replied:
Yea I get it's not ideal from purity standpoint, but we're not the first to [do this](https://marketplace.visualstudio.com/items?itemName=msjsdiag.debugger-for-chrome).
It's supported by Github and the VSCode marketplace.

jpoon commented on README.md:
```diff
@@ -393,29 +437,43 @@ If you are use to using vim-commentary you are probably use to using `gc` instea
-## Special Shoutouts to Cool Contributors
+## F.A.Q.
+
+### `j`, `k` and others keys don't repeat when I hold them down.
```
You had this covered in the Mac setup? Maybe delete it from here.

cobbweb replied:
I thought it might be be worth leaving it duplicated for a little while? I know I've linked to this section before getting friends up-to-speed. Happy to remove though.

jpoon replied:
Yes. Please delete.

jpoon commented on README.md:
```diff
@@ -27,12 +43,32 @@ Ask us questions, talk about contributing, or just say hi on [Slack](https://vsc
+## Contents
 
-## Configuring VSCodeVim
+* [Quick example settings](#quick-example-settings)
```
I think `quick example settings` would below under the `settings` category.

Mac/Windows setup could be under it's own `setup`/getting started category.

cobbweb replied:
Yea good idea, I knew this section was 'off'. I'll move the quick settings into Settings, and rename this to 'Getting started', and might even include some links to some vim learning resources?

rebornix commented on README.md:
```diff
@@ -1,19 +1,35 @@
 VSCodeVim is a [Visual Studio Code](https://code.visualstudio.com/) extension that enables Vim keybindings, including:
 
-* Modes: normal, insert, command, visual, visual line, visual block (with `useCtrlKeys`, see below)
+* Modes: normal, insert, command-line, visual, visual line, visual block
```
we should use `command` as it's referring to ex-command.

cobbweb replied:
I changed it in case there was any confusion with the next point: "Command combinations". It's also called [command-line mode in the vim docs](http://vimdoc.sourceforge.net/htmldoc/cmdline.html#Command-line).

rebornix commented on README.md:
```diff
@@ -51,14 +87,14 @@ Below is an example of a [settings.json](https://code.visualstudio.com/Docs/cust
         {
-           "before":["<C-n>"],
-           "after":[],
+            "before":["<C-n>"],
```
Let's keep using 2 spaces.

rebornix commented on README.md:
```diff
@@ -68,70 +104,110 @@ Below is an example of a [settings.json](https://code.visualstudio.com/Docs/cust
-       ]
-   }
+    {
+        "before": [":"],
```
same here, I suppose it's changed by some indenting or formatting?

cobbweb replied:
The indentation was all whacky, I re-indented it all so it was consistent. (I think there was 3-spaces here)

rebornix commented:
Just a few tiny suggestions, it looks good to me and it's way way awesome.

jpoon commented:
Also, while you are at this, mind taking care of this? https://github.com/VSCodeVim/Vim/issues/1466

cobbweb commented:
Ok pushed a few extra bits:

1. Added a Getting started section to house the OS-specific setup. I also add some new words, but I think it's just waffling and that I should probably remove it again?
2. Added a small sub-section regarding Vim compatibility, with another link to the roadmap (there's already one in the features list)
3. Removed the duplicate info about key-repeat on Macs

jpoon commented:
One last thing, mind squashing?

xconverge commented:
We can squash when we merge too...not a big deal

xconverge commented on README.md:
```diff
@@ -27,12 +43,58 @@ Ask us questions, talk about contributing, or just say hi on [Slack](https://vsc
+
+### Vim compatibility
+
+All common Vim commands are supported. For an in-depth look at what Vim features are supported, check out the [roadmap](ROADMAP.md). Vimscript isn't supported, so you aren't able to load your `.vimrc` or use `.vim` plugins. You have to replicated this using our [Settings](#settings) and [Emulated plugins](#emulated-plugins).
```
small grammar issue in last sentence "you have to replicate these"

cobbweb replied:
Fixed.

johnfn commented:
Sorry, not going to have the time to review this this weekend, but I'm sure I can just merge in changes directly if I notice anything is horribly wrong. :) (Which I doubt.)

xconverge commented:
Thanks a ton! This looks great and is definitely a solid start to build on!

mightyiam commented:
Thank you so much!

Enkdress commented on README.md:
```diff
@@ -393,29 +446,34 @@ If you are use to using vim-commentary you are probably use to using `gc` instea
+* `gq` - on a visual selection - Reflow and wordwrap blocks of text, preserving commenting style. Great for formatting documentation comments.
+* `gc` - which adds another cursor on the next word it finds which is the same as the word under the cursor.
+* `af` - a command that I added in visual mode, which selects increasingly large blocks of text. e.g. if you had "blah (foo [bar 'ba|z'])" then it would select 'baz' first. If you pressed `af` again, it'd then select [bar 'baz'], and if you did it a third time it would select "(foo [bar 'baz'])".
+* `gh` - another custom VSCodeVim command. This one is equivalent to hovering your mouse over wherever the cursor is. Handy for seeing types and error messages without reaching for the mouse!
```
Can I have this feature in nvim with any plugin or something like that

J-Fields replied:
Haven't used nvim seriously for a while but this might be useful: https://neovim.io/doc/user/lsp.html
